core/src/main/java/de/danoeh/antennapod/core/storage/PodDBAdapter.java

```diff
@@ -484,7 +387,12 @@ public void setFeedItemFilter(long feedId, List<String> filterValues) {
      * @return the id of the entry
      */
     public long setImage(FeedImage image) {
-        db.beginTransaction();
+        boolean startedTransaction = false;
+        if(false == db.inTransaction()) {
+            db.beginTransaction();
+            startedTransaction = true;
+        }
```

Comment:
Not sure if this is actually necessary. Added this when I tried to fix a bug that turned out to be caused by another issue. But this should not hurt either. (Actually, there are some threads on SO that claim Android's sqlite does not support nested transactions and that one should keep track of them)